tools/socktap/dcc_passthrough.cpp

```diff
@@ -43,7 +44,9 @@ void DccPassthrough::request(const dcc::DataRequest& request, std::unique_ptr<Ch
         const_buffers[i] = asio::buffer(buffers_[i]);
     }
     auto bytes_sent = socket_.send(const_buffers);
-    std::cout << "sent packet to " << request.destination << " (" << bytes_sent << " bytes)\n";
+    std::ostringstream ostream;
+    ostream << "sent packet to " << request.destination << " (" << bytes_sent << " bytes)\n";
+    std::cout << ostream.str();
```

Comment:
Why is this beneficial? From my point of view, this causes an additional memory allocation and thus this should be justifiable.

Reply:
When Vanetza shares stdout with other threads that manipulate the flags for stdout, it can happen (although rarely and unpredictable) that the `std::hex` flag set in the streaming operator overload of `MacAddress` is not properly cleared. One could probably move the special treatment to the streaming operator overload.

If you don't see that as a problem Socktap should care about, feel free to omit this commit.

Reply:
Okay, but socktap is not multi-threaded ;-) If this is going to change, we will need some more sophisticated logging anyway.